### ISSUE TITLE
Stop watch job

### ISSUE THREAD

**Issue by TimMeissner** [jhildenbiddle/css-vars-ponyfill#35]
Hey, 

is it possible to stop the watch task once it has been started? In my app the user can apply new themes as he likes and we use the watch job to allow view encapsulation / component styles (the component styles are added on the fly as a style tag to the head when the component is rendered for the first time). So when we change the theme i currently have the problem that i can't change the variables or start a new task because the old one is still running.

**Comment by jhildenbiddle** [comment]
Hi @TimMeissner.

The ponyfill does not provide a way to disconnect the [`options.watch`](https://github.com/jhildenbiddle/css-vars-ponyfill#optionswatch) MutationObserver, but it should. I've added this enhancement request to the backlog and will get to it in the next few days.

In the meantime, depending on how you are swapping themes you shouldn't need to disconnect/reconnect the `options.watch` MutationObserver to apply new themes. Disconnecting and reconnecting the MutationObserver should really only be necessary if you are changing the ponyfill's `watch` configuration (e.g. `rootElement`, `include`, `exclude` etc).

* If applying a new theme means modifying a collection of CSS custom property values, just pass the new values to the ponyfill using [`options.variables`](https://github.com/jhildenbiddle/css-vars-ponyfill#optionsvariables). The ponyfill is smart enough to update both modern and legacy browsers, and the MutationObserver is smart enough to know that these changes do not require a subsequent ponyfill call.

   ```javascript
   // Enable MutationObserver
   cssVars({
     watch: true
   });

   // Do stuff...

   // Add/modify CSS custom property value
   cssVars({
     variables: {
       '--my-color': 'red'
     }
   });
   ``

* If applying a new theme means injecting or enabling/disabling a stylesheet, then just inject/enable/disable as needed and the [`options.watch`] MutationObserver should detect the changes and re-run the ponyfill automatically. Note that the ponyfill will **not** detect [`characterData`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserverInit/characterData) mutations within `<style>` nodes.

If the tips above don't help, I may be able to offer more insight with additional details about your app. If possible, consider creating an online demo with https://codesandbox.io/ or https://stackblitz.com/.

**Comment by jhildenbiddle** [comment]
If the ponyfill is taking 3-5 seconds on the initial call, you must be processing **a lot** of CSS. If you don't mind me saying so, that's the first place I'd start. Are you able to share your CSS, how the files are organized and loaded into the app, and anything else related that may be of interest (for example, if you're injecting route-based CSS, processing shadowDOM elements, etc.)? If so, we may be able to speed up the initial ponyfill call and (hopefully) let the ponyfill just do its thing.

In the meantime, I'll work on adding the ability to disconnect/reconnect the ponyfill's MutationObserver.

**Comment by TimMeissner** [comment]
Thanks for the quick help! 
I will test the new feature tomorrow morning and also investigate the amount of CSS. I hope i can setup a stackblitz or smth. like that. 
Please don't close this thread yet, i will get back to you.

**Comment by TimMeissner** [comment]
The good news is that i don't even need to disconnect the observer now (i think because you added that it's possible to change the settings of the watch task). The problem disappeared with the newest version.
Big thanks! 👍 

So indeed we have a lot of CSS (angularjs + Angular 6 hybrid app): 
chunk {styles} styles.js, styles.js.map (styles) 1.97 MB [initial] [rendered]
with 11401 rules and even more selectors.
Half of that is by libraries and the other half is our own css code. I doubt that it is possible to reduce the amount of css right now, not until we have finished migrating from angularjs to angular 6 (and therefore from global styles to mostly component styles).
The problem is solved, so you can close this.

**Comment by jhildenbiddle** [comment]
You bet. Glad the update worked for you.

Good luck with the transition. 😄
